### DISCUSSION TITLE
Localization

### DISCUSSION THREAD

**Discussion by gapfranco** [filamentphp/filament#2533] [Help]
How can I change the locale at run-time?  I tried Config::set('app.locale', $locale); and app()->setLocale($locale); It keeps using the settings in filament config.file.

Someone mentioned using:

    Filament::serving(function () {
        Config::set('app.locale', $locale);
        Carbon::setLocale($locale);
    });

but I cannot figure out how to use it. What is Filament::serving?

Any help would be appreciated.

**Comment by danharrin** [comment]
That would work in a service provider, or maybe a middleware if your `$locale` is dynamic per-user?

Make sure to register the middleware in config/filament.php

**Reply by gapfranco** [reply]
Yes, in my application it's dynamic per user. The users chooses the language in a link and the locale is saved in the session and used in blade templates.
I'm trying to use the same idea in a Filament app.

**Reply by danharrin** [reply]
Do it in middleware then :)

**Reply by gapfranco** [reply]
Ok. Thank you very much for the tip.
